rpcs3/Emu/Cell/Modules/cellMic.cpp

```diff
@@ -258,6 +258,29 @@ inline void microphone_device::variable_byteswap(const void* src, void* dst)
 	}
 }
 
+inline u32 microphone_device::convert_16_bit_pcm_to_float(const std::vector<u8>& buffer, usz num_bytes)
+{
+	constexpr usz float_buf_size = 2 * inbuf_size;
+	static_assert((float_buf_size % sizeof(u32)) == 0);
+
+	float_buf.resize(float_buf_size, 0);
+	ensure(num_bytes * 2 <= float_buf.size());
+
+	const be_t<s16>* src = reinterpret_cast<const be_t<s16>*>(buffer.data());
```

Comment:
This produces a "dereferencing type-punned pointer" warning on gcc.
Maybe we have to cast the buffer ptr to void...

```suggestion
	const be_t<s16>* src = reinterpret_cast<const be_t<s16>*>(static_cast<const void*>(buffer.data()));
```